src/main/java/net/imglib2/roi/labeling/OverlappingLabels.java

```diff
@@ -0,0 +1,194 @@
+import net.imglib2.img.Img;
+import net.imglib2.img.array.ArrayImgs;
+import net.imglib2.loops.LoopBuilder;
+import net.imglib2.type.numeric.integer.LongType;
+import net.imglib2.type.numeric.integer.UnsignedIntType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.util.Util;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+public class OverlappingLabels< T extends Comparable< T > >
+{
+
+	private Img< UnsignedIntType > overlapMatrix;
+
+	private List< T > labelList;
+
+	private Img< LongType > labelSizeImg;
+
+	private Img< DoubleType > relativeOverlapMatrix;
+
+	public OverlappingLabels( RandomAccessibleInterval< LabelingType< T > > labeling )
+	{
+
+		FragmentLabelRegions labelRegions = new FragmentLabelRegions( labeling );
+
+		// List of existing labels
+		// NB: this triggers labelRegions computation
+		labelList = new ArrayList<>( labelRegions.getExistingLabels() );
+		Collections.sort( labelList );
+
+		// Create 2D matrix image
+		FinalInterval matrixDims = new FinalInterval( labelList.size(), labelList.size() );
+		overlapMatrix = Util.getSuitableImgFactory( matrixDims, new UnsignedIntType() ).create( matrixDims );
+
+		// Sum up sizes of single fragments
+		RandomAccess< UnsignedIntType > randomAccess = overlapMatrix.randomAccess();
+		UnsignedIntType sizeType = new UnsignedIntType();
+		LabelingMapping< T > labelingMapping = Labelings.getLabelingMapping( labeling );
+		for ( int i = 0; i < labelingMapping.numSets(); i++ )
+		{
+			Set< T > currentLabels = labelingMapping.labelsAtIndex( i );
+			long size = labelRegions.getFragmentSizeForIndex( i );
+			if ( size > 0 )
+			{
+				sizeType.set( size );
+				for ( T a : currentLabels )
+				{
+					randomAccess.setPosition( labelList.indexOf( a ), 0 );
+					for ( T b : currentLabels )
+					{
+						randomAccess.setPosition( labelList.indexOf( b ), 1 );
+						randomAccess.get().add( sizeType );
+					}
+				}
+			}
+		}
+
+		labelSizeImg = ArrayImgs.longs( labelList.size() );
+		Cursor< LongType > cursor = labelSizeImg.localizingCursor();
+		while ( cursor.hasNext() )
+		{
+			cursor.next().set( labelRegions.getLabelRegion( labelList.get( cursor.getIntPosition( 0 ) ) ).size() );
+		}
+
+		relativeOverlapMatrix = Util.getSuitableImgFactory( matrixDims, new DoubleType() ).create( matrixDims );
+		IntervalView< LongType > divisor = Views.addDimension( labelSizeImg, 0, labelList.size() - 1 );
+
+		LoopBuilder.setImages( overlapMatrix, divisor, relativeOverlapMatrix ).forEachPixel( ( a, b, c ) -> c.set( ( double ) a.get() / b.get() ) );
+	}
+
+	/**
+	 * Get the confusion/co-occurrence matrix.
+	 * 
+	 * Values indicate the number of pixels that are in common between the label
+	 * at position x and the label at position y. For the mapping of dimension
+	 * indices to labels, see {@link OverlappingLabels#getIndexedLabels()}.
+	 * 
+	 * @return the co-occurrence matrix
+	 */
+	public Img< UnsignedIntType > getMatrixImg()
```

Comment:
I would prefer to return RandomAccessibleInterval here. Is there any reason not to?

Reply:
`overlapMatrix` is an `Img<UnsignedIntType>`, so do you also mean to change the type of the private field?

Otherwise, why not return `Img` if we can fulfill both contracts of `RandomAccessibleInterval` and `IterableInterval`? I would save a call to `Views.iterable` downstream, no?

And we don't require any non-zero offset for the overlap matrix anyhow, do we?

So I don't see compelling arguments to return `RandomAccessibleInterval` here, but I'll change it if you prefer so.

Reply:
The only reason is to be free to change internals later without breaking API.
But I don't feel very strongly about it.

Reply:
@tpietzsch that's a fair point. I changed the return type to `RandomAccessibleInterval` and renamed the method to `getMatrix` instead of `getMatrixImg`.